pubsub/drivertest/drivertest.go

```diff
@@ -245,6 +249,52 @@ func testSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 }
 
+// testReceiveSend checks that a receive will still return even if the next
+// message comes in after it was called.
+func testReceiveSend(t *testing.T, newHarness HarnessMaker) {
+	// Set up.
+	ctx := context.Background()
+	h, err := newHarness(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer h.Close()
+	top, sub, cleanup, err := makePair(ctx, h, t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	// Start a receive in the background.
+	gotChan := make(chan *pubsub.Message)
+	go func() {
+		m, err := sub.Receive(ctx)
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		m.Ack()
+		gotChan <- m
+	}()
+
+	// Wait for Receive to block, with any luck.
+	time.Sleep(100 * time.Millisecond)
+
+	// Send a message.
+	want := &pubsub.Message{Body: []byte(fmt.Sprintf("%d", rand.Int()))}
+	if err := top.Send(ctx, want); err != nil {
+		t.Fatal(err)
+	}
+
+	// The same message should come through from gotChan.
+	got := <-gotChan
+	gotStr := string(got.Body)
```

Comment:
I think you can just use `cmp.Equal` here instead of casting, then `%v` in the error.

Reply:
Changed it to use `cmp.Equal` but `%v` would print out the byte slices literally so I'd like to leave it as `%s`.